### ISSUE TITLE
Accessibility: Conditional key handlers

### ISSUE THREAD

**Issue by jurokapsiar** [microsoft/fluent-ui-react#1502]
### Problem description
There is not a clear separation between behaviors and components if the action executed on the component should be different based on props or the event. See https://github.com/stardust-ui/react/pull/1488/files#r293815521

### Proposed solution
Allow conditional keyboard handlers that would only execute an action if a condition based on the props or event is met

**Comment by jurokapsiar** [comment]
Recommendation is to conditionally (based on props) add only relevant key handlers.
